pkg/netlight/network.go

```diff
@@ -0,0 +1,378 @@
+package netlight
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"slices"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/cache"
+	"github.com/threefoldtech/zos/pkg/netlight/bridge"
+	"github.com/threefoldtech/zos/pkg/netlight/ifaceutil"
+	"github.com/threefoldtech/zos/pkg/netlight/ipam"
+	"github.com/threefoldtech/zos/pkg/netlight/macvlan"
+	"github.com/threefoldtech/zos/pkg/netlight/namespace"
+	"github.com/threefoldtech/zos/pkg/netlight/options"
+	"github.com/threefoldtech/zos/pkg/netlight/resource"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	NDMZBridge    = "br-ndmz"
+	NDMZGw        = "gw"
+	mib           = 1024 * 1024
+	ipamLeaseDir  = "ndmz-lease"
+	DefaultBridge = "zos"
+)
+
+var (
+	NDMZGwIP = &net.IPNet{
+		IP:   net.ParseIP("100.127.0.1"),
+		Mask: net.CIDRMask(16, 32),
+	}
+)
+
+type networker struct {
+	ipamLease string
+}
+
+var _ pkg.NetworkerLight = (*networker)(nil)
+
+func NewNetworker() (pkg.NetworkerLight, error) {
```

Comment:
is it common to add `er` to the `struct` in `zos`?
because `er` suffix usually used for `interface`, not `struct`